EVENT-508

dougworner opened the pull request:
Add event search loading indicator to landing page.

OzzieOrca commented:
I don't think we need to block user interaction during searching. At the end of the debounce, it prevents the user from changing their search until the server comes back with a response.

I was thinking something more like this:

<img width="961" alt="screen shot 2017-08-17 at 12 52 10 pm" src="https://user-images.githubusercontent.com/756501/29430935-06d415e4-834b-11e7-947b-74d29dfbd4b8.png">


```html
<div class="text-center">
  <h2>Searching events…</h2>
  <i class="fa fa-spinner fa-spin fa-4x"></i>
</div>
```

Idk, what do you think? You could try some other styles or throw it in a panel/well if that looks any better. Idk

dougworner commented:
I tested your code.  How would we control it so it doesn't spin continuously? As soon as I bring up the site the indicator is spinning.

I did some other testing as well you might find interesting.  I replaced $rootScope with $scope and it appears now that the user is able to interact.  I added more places to clear out the value since I noticed when I type in a value and then backspaced it was not being cleared.

What do you think?

OzzieOrca commented on app/scripts/controllers/landing.js:
```diff
@@ -29,8 +29,12 @@ angular.module('confRegistrationWebApp')
+      if(!val){
+        $scope.loadingMsg = '';
+        return; }
       if(val.length < 2){
```
Can you consolidate this with the if above? Seems like they do the same thing.

OzzieOrca commented:
Umm you can toss the `fa-spin` class with `ng-class` if you don't want it to spin some of the time. Idk what you have in mind.

I guess I usually try to not add overlays unless they are really necessary. My idea would be to show the current initial content (Register for an event To register for an event, search...) when the user hasn't typed anything or has deleted their search, show the loading indicator in place of it during a search and then show the results when they come back.

dougworner commented:
Let me know if this is closer to what you were thinking.  I left fa-spin in for now.

dougworner commented:
Made one small change.

OzzieOrca commented on app/views/landing.html:
```diff
@@ -32,6 +32,10 @@ <h1 translate>Search for an event</h1>
+  <h2>{{loadingMsg}}</h2>
+  <i class="fa fa-spinner fa-spin fa-4x"></i>
+</div>
 <div class="container spacing-above-md" ng-if="!searchComplete">
```
I think we should hide this when showing the loading indicator.
Could do: `ng-if="!loadingMsg && !searchComplete"`

OzzieOrca commented on app/views/landing.html:
```diff
@@ -32,6 +32,10 @@ <h1 translate>Search for an event</h1>
   </div>
 </div>
+<div class="text-center" ng-if="loadingMsg">
+  <h2>{{loadingMsg}}</h2>
```
I think you can just hardcode `Searching Events&hellip;` here since the message never changes.

You can make it translatable by doing `<h2 translate>Searching Events&hellip;</h2>`.

Then `loadingMsg` could just be a flag (true/false). Can you think of a more descriptive variable name for it that implies it is for searching? Maybe `loadingSearchResults` or something.

OzzieOrca commented on app/scripts/controllers/landing.js:
```diff
@@ -29,23 +29,24 @@ angular.module('confRegistrationWebApp')
         }
       }).then(function(response){
+        $scope.loadingMsg = '';
         if(angular.isDefined($scope.searchVal) && val === $scope.searchVal) {
```
Disabling the loading indicator should be done in here cuz that's where we know we have results for the current search query and that `$scope.searchResults` has been set.

OzzieOrca commented on app/scripts/controllers/landing.js:
```diff
@@ -29,8 +29,9 @@ angular.module('confRegistrationWebApp')
     $scope.eventSearch = _.debounce(function(val) {
-      if(!val){ return; }
-      if(val.length < 2){
+      $scope.loadingSearchResults = true;
```
If you move this to after the if statement that checks the length, you shouldn't have to deal with it in the if.
